### PR TITLE
Add support for binary and hex files

### PR DESCRIPTION
**Pull request by simonbothen-radinn**
In some cases it is more convinient to use a binary or hex file directly instead of using a zip file.

We can check the extention and do the appropriate action.

### CODE REVIEW

**Comment by manualexSP** [comment]
I will have a look and do some tests in the next period. Thank you for contributing

File: android/src/main/java/com/pilloxa/dfu/RNNordicDfuModule.java [hunk 1]
```diff
@@ -38,7 +38,13 @@ public void startDFU(String address, String name, String filePath, Promise promi
             starter.setDeviceName(name);
         }
         starter.setUnsafeExperimentalButtonlessServiceInSecureDfuEnabled(true);
-        starter.setZip(filePath);
+
+        if (filePath.endsWith(".bin") || filePath.endsWith(".hex")) {
```
Comment (mmesarosiuSP) [inline android/src/main/java/com/pilloxa/dfu/RNNordicDfuModule.java:42]: On iOS the extension check is case insensitive, but on Android it isn't.  Wouldn't it be better to check against the filePath's toLowerCase()?

**Comment by mmesarosiuSP** [comment]
I've added your [repo's branch](https://github.com/simonbothen-radinn/react-native-nordic-dfu/tree/master_binary_support) as npm dependency, however on my Android 12 device it was crashing with your branch. I'm assuming because you don't have the Android 12  updates from this [PR](https://github.com/Salt-PepperEngineering/react-native-nordic-dfu/pull/2).
Instead I applied these changes on our repo and installed it as a local npm package and it was working properly.

**Comment by simonbothen-radinn** [comment]
> I've added your [repo's branch](https://github.com/simonbothen-radinn/react-native-nordic-dfu/tree/master_binary_support) as npm dependency, however on my Android 12 device it was crashing with your branch. I'm assuming because you don't have the Android 12 updates from this [PR](https://github.com/Salt-PepperEngineering/react-native-nordic-dfu/pull/2). Instead I applied these changes on our repo and installed it as a local npm package and it was working properly.

Thanks for the feedback. Will look into the above next week.

**Comment by simonbothen-radinn** [comment]
> I've added your [repo's branch](https://github.com/simonbothen-radinn/react-native-nordic-dfu/tree/master_binary_support) as npm dependency, however on my Android 12 device it was crashing with your branch. I'm assuming because you don't have the Android 12 updates from this [PR](https://github.com/Salt-PepperEngineering/react-native-nordic-dfu/pull/2). Instead I applied these changes on our repo and installed it as a local npm package and it was working properly.

Rebased to remotes HEAD.

**Comment by manualexSP** [comment]
Moved your implementation on this MR https://github.com/Salt-PepperEngineering/react-native-nordic-dfu/pull/6

**Pull request closed without merging**